### PR TITLE
Implement simple FIT parser and link with Drag&Drop interaction

### PR DESCRIPTION
**Pull request by lbesson**
Solves #1724

### CODE REVIEW

**Comment by asaunier** [comment]
\o/

File: src/js/fit/FIT.js [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+    return FormatType.ARRAY_BUFFER;
+  }
+
+  readProjection() {
+    return this.dataProjection;
+  }
+
+  addEndian(littleEndian, bytes) {
+    let result = 0;
+    if (!littleEndian) bytes.reverse();
```
Comment (brunobesson) [inline src/js/fit/FIT.js:76]: I dislike `if`s without bracket 😞

File: src/js/fit/FIT.js [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+      const { nextIndex, messageType, message } = this.readRecord(blob, loopIndex);
+      loopIndex = nextIndex;
+
+      if (messageType === MESSAGE_TYPE.DATA_RECORD && message.longitude && message.latitude) {
+        geometry.appendCoordinate([
+          message.longitude,
+          message.latitude,
+          message.altitude || 0,
+          message.timestamp?.getTime() || 0,
+        ]);
```
Comment (brunobesson) [inline src/js/fit/FIT.js:383]: We could also ignore points without elevation, unless none has elevation

File: src/js/fit/FIT.js [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+     * and cannot be used for local message types 4–15.
+     */
+    const localMessageType = isCompressedTimestampDataMessage
+      ? (recordHeader & COMPRESSED_LOCAL_MESSAGE_NUMBER_MASK) >> 5
+      : recordHeader & LOCAL_MESSAGE_TYPE_MASK;
+
+    /** Get the message type from the definitions */
+    const messageType = this.messageTypes[localMessageType];
+    if (!messageType) {
+      throw new Error('Missing message definition for local message number');
```
Comment (brunobesson) [inline src/js/fit/FIT.js:231]: sometimes you throw errors, sometimes you console.log and return empty array.
what will happen here on error, where is it caught?

**Review by brunobesson — COMMENTED** [review]
We should handle fit files in file input too (and add missing kml)

File: src/js/fit/FIT.js (hunk 1 above)
Comment (lbesson) [reply src/js/fit/FIT.js:76]: yeah, bad copy paste

File: src/components/map/OlMap.vue [hunk 4]
```diff
@@ -485,12 +486,23 @@ export default {
...
+        return null;
+      }
+    },
 
-      features.map(this.setDocumentGeometryFromFeature);
-      this.fitMapToDocuments(true);
+    setDocumentGeometryFromGeoFile(file) {
+      for (const format of [new ol.format.GPX(), new FIT(), new ol.format.KML()]) {
+        const features = this.tryReadFeaturesFromGeoFile(file, format, { featureProjection: 'EPSG:3857' });
+        if (features && features.length) {
```
Comment (brunobesson) [inline src/components/map/OlMap.vue:500]:
```suggestion
        if (features?.length) {
```

**Comment by lbesson** [comment]
there is an issue with the timestamps

**Comment by brunobesson** [comment]
Please merge when you are ok, it's fine for me 👏

**Pull request merged**